src/Uart/Uart.c

```diff
@@ -268,24 +268,26 @@ Uart_getRxFifoCount(Uart *const uart)
 	return count;
 }
 
-static inline void
-handleRxInterrupt(Uart *const uart)
+static inline bool
+handleRxInterrupt(Uart *const uart, int* const errCode)
 {
 	uint8_t data = (uint8_t)uart->reg->rhr;
 
 	if (uart->rxFifo == NULL) {
 		disableRxIrq(uart);
-		return;
+		return true;
 	}
 
-	ByteFifo_push(uart->rxFifo, data);
+	const bool retValue = ByteFifo_push(uart->rxFifo, data) ? true : returnError(errCode, Uart_ErrorCodes_Rx_Fifo_Full);
```

Comment:
Why don't we return immediatelly after error occurence?

Reply:
We don't do that because handlers wouldn't be called in this scenario. The idea of this software is to use handlers to pass information to upper layers of code that something of interest was received. Then, we can pull from the FIFO to read the data and free some space in it. On the other hand, If we would return immediately the information could be lost and the FIFO could be full all the time. Thus, we would never recover from this error.

Reply:
"The idea of this software is to use handlers to pass information to upper layers of code that something of interest was received." - OK, but the callbacks are invoked in an incorrect state.
target character callback is called... but the character is not in the fifo... so we have to handle another situation there. So we are propagating the failure, instead of isolating it. If this is in the interrupt, maybe error handler should pass it to the user code? What do you think?